### DISCUSSION TITLE
Save energy while Wifi is disconnected

### DISCUSSION THREAD

**Discussion by mi-schi** [arendst/Tasmota#15401] [Q&A] [answered]
Hello,

is there a possibility to save energy while the Wifi is disconnected? I turn off the router every night. Currently I set the WifiConfig to 5 = wait until selected AP is available again without rebooting for the tasmota plugs. Sadly one plug needs 1.2 watt while waiting. When Wifi is connected, the plug consume only 0.5 watt.

I program a ESP too and use this code to save a lot of energy:
```
void loop() {
  if(WiFi.status() != WL_CONNECTED) {
    ESP.deepSleep(NO_WIFI_SLEEP_MINUTES * 6e7);

    return;
  }

  // ...

```
So my ESP sleep 15 minutes and than try to reconnect. It is okay for my use case. The tasmota DeepSleep routine is not the solution I think. Is there also a possibility to set a deep sleep timer while waiting for Wifi?

**Comment by ascillato** [comment]
> The tasmota DeepSleep routine is not the solution I think.

Why not? Is exactly that. You can use timers and rules to trigger DeepSleep. All the logic can be in a Tasmota rules to decide to enter in DeepSleep or not. As your use-case can not be applied generally (a light can not be unresponsive for 15 minutes each night), it is better to be configured using Tasmota timers and rules.

**Reply by barbudor** [reply]
Tasmota DeepSleep is not the right answer
Tasmota DeepSleep has been designed for battery powered sensor. When Tasmota wakes up from DeepSleep, it tries to connect to Wifi and MQTT and when it gets connectes, it publish the sensors to MQTT and goes back to sleep.
So if Wifi is not available, it tries to connect almost endlessly (I think there is a timeout and then it goes back to sleep)

Also nothing can happen when the ESP is in Deepsleep


There are other sleep modes on the ESP that would be more efficient (like "modem sleep" which turns the power hungry radio part off) but I would expect implementing that in Tasmota a little complex.

**Reply by ascillato** [reply]
Modem sleep is already implemented and working. The command `sleep 50` (Tasmota's default) already does that and it does that in a dynamic way (turns on modem sleep only when it is possible and in between Wi-Fi Beacons <- DTIM parameter of routers). This is how Tasmota reduces its energy consumption and also continues being responsive.
Tasmota already uses a low amount of energy, but if the OP wants to shutdown the device at night because they are turning off the wifi router, they can use deepsleep. Deepsleep is for a very specific use-case. In that mode, the ESP consumes very low energy but it is in a shutdown mode, and won't answer any command until it wakes up. DeepSleep is meant for sensors and not for controllers. Due to that, if the user wants to use it, it can be configured using rules.

**Comment by mi-schi** [comment, marked as answer]
I set sleep already to 50, so the consumption is reduced from 0.7 watt to 0.3 watt.
With the tested plug I have 0.7 watt without Wifi, 0.3 watt with Wifi. I try to set the following rule:
```
Rule1 ON wifi#disconnected DO DeepSleepTime 900 ENDON ON wifi#connected DO DeepSleepTime 0 ENDON
```
But I see that this plug has a very old version 6.5.0.16
I try to update the plug and than try to use the rule.

**Reply by barbudor** [reply]
Did you connected GPIO16 to RST?
This is mandatory.
Read the doc

**Reply by mi-schi** [reply]
After I upgrade to the last release with 8 intermediate steps ;) I test the given rule. The first seconds after Wifi is disconnected the consumption drops to 0 watt, then to 0.2 watt. But, as you mentioned the ESP didn't wake up because of the missing connection from RST to GPIO16 (8) XPD_DCDC, see:
![ESP8266_pinout](https://user-images.githubusercontent.com/8821732/164675842-5a42a730-a4b1-4877-90b1-256c80317401.jpg)

The problem, this pin is not accessible I think, or I'm wrong?
This one here:
https://electronics.stackexchange.com/questions/404875/im-trying-to-identify-this-wifi-microcontroller-without-any-luck

Is the same like mine:
![20220422_110635](https://user-images.githubusercontent.com/8821732/164676489-475cdf9b-06ff-4ff7-aca0-a2ece07f51c8.jpg)
![20220422_110645](https://user-images.githubusercontent.com/8821732/164676492-3f313bc2-e9ba-4a8e-bb28-59ec3d1b5c6c.jpg)
![20220422_110723](https://user-images.githubusercontent.com/8821732/164676496-15012724-cf4b-429b-996c-e4529ca5c6c0.jpg)

**Reply by sfromis** [reply]
That board is indeed not designed for what you're trying to do. If you really, really want to proceed, you can lift off the metal can covering the ESP chip inside, and solder to the tiny pins on the chip.
